data-serving/data-service/src/index.ts

```diff
@@ -27,6 +29,10 @@ apiRouter.put('/cases/:id([a-z0-9]{24})', caseController.update);
 apiRouter.delete('/cases/:id([a-z0-9]{24})', caseController.del);
 app.use('/api', apiRouter);
 
+// API documentation.
+const swaggerDocument = YAML.load('./api/dataserver.api.yaml');
+app.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerDocument));
```

Comment:
So cool. Do you think documentation is the primary outcome here? Do we want to accomplish anything else based on the spec?

Reply:
I think documentation is the low-hanging fruit, but I think using the spec to enforce the API would be even nicer! I think we have a few options, would love feedback/thoughts:

- At a minimum I'd like to validate requests & responses, but still evaluating the different libraries for this (there are a couple and I want to play with an example to see which is better, like https://github.com/cdimascio/express-openapi-validator vs. https://github.com/urma/express-openapi vs https://github.com/isa-group/oas-tools).

- We could potentially use the tags in code as part of RBAC

- Trying to figure out if any of the validation libs generate nice TS bindings to help us construct requests or responses -- presumably yes, since they can validate but want to dig in and see

Reply:
> At a minimum I'd like to validate requests & responses

Validation SGTM. Will have to make sure the overall behavior is coherent with both request validation and data model validation in the mix.

> We could potentially use the tags in code as part of RBAC

Yeah, I'll uninformedly bet that with enough elbow grease we could manually operate on the tags in middleware. Will be interesting to see if something like this works well enough for us to use.

> Trying to figure out if any of the validation libs generate nice TS bindings to help us construct requests or responses

SG